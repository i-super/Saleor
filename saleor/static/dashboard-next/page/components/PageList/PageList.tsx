import VisibilityIcon from "@material-ui/icons/Visibility";
import blue from "material-ui/colors/blue";
import IconButton from "material-ui/IconButton";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableFooter,
  TableHead,
  TableRow
} from "material-ui/Table";
import * as React from "react";
import { Link } from "react-router-dom";

import Skeleton from "../../../components/Skeleton";
import TablePagination from "../../../components/TablePagination";
import { PageListQuery } from "../../../gql-types";
import i18n from "../../../i18n";

interface PageListComponentProps {
  pages?: PageListQuery["pages"]["edges"];
  pageInfo: PageListQuery["pages"]["pageInfo"];
  handlePreviousPage();
  handleNextPage();
  onEditClick(id: string);
  onShowPageClick(slug: string);
}

const decorate = withStyles({
  link: {
    color: blue[500],
    cursor: "pointer",
    textDecoration: "none"
  },
  textRight: {
    textAlign: "right"
  }
});

export const PageListComponent = decorate<PageListComponentProps>(
  ({
    classes,
    handlePreviousPage,
    handleNextPage,
    onEditClick,
    onShowPageClick,
    pages,
    pageInfo
  }) => (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell>{i18n.t("Name", { context: "object" })}</TableCell>
          <TableCell>{i18n.t("Url", { context: "object" })}</TableCell>
          <TableCell>{i18n.t("Visibility", { context: "object" })}</TableCell>
          <TableCell className={classes.textRight}>
            {i18n.t("Actions", { context: "object" })}
          </TableCell>
        </TableRow>
      </TableHead>
      <TableFooter>
        <TableRow>
          <TablePagination
            colSpan={4}
            hasNextPage={pageInfo ? pageInfo.hasNextPage : false}
            onNextPage={handleNextPage}
            hasPreviousPage={pageInfo ? pageInfo.hasPreviousPage : false}
            onPreviousPage={handlePreviousPage}
          />
        </TableRow>
      </TableFooter>
      <TableBody>
        {pages === undefined || pages === null ? (
          <TableRow>
            <TableCell>
              <Skeleton />
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
            <TableCell className={classes.textRight}>
              <IconButton disabled>
                <VisibilityIcon />
              </IconButton>
            </TableCell>
          </TableRow>
        ) : pages.length > 0 ? (
          pages.map(edge => (
            <TableRow key={edge.node.id}>
              <TableCell
                onClick={onEditClick(edge.node.id)}
                className={classes.link}
              >
                {edge.node.title}
              </TableCell>
              <TableCell>{`/${edge.node.slug}`}</TableCell>
              <TableCell>
                {edge.node.isVisible
                  ? i18n.t("Published", { context: "object" })
                  : i18n.t("Not published", { context: "object" })}
              </TableCell>
              <TableCell className={classes.textRight}>
                <IconButton onClick={onShowPageClick(edge.node.slug)}>
                  <VisibilityIcon />
                </IconButton>
              </TableCell>
            </TableRow>
          ))
        ) : (
          <TableRow>
            <TableCell colSpan={3}>{i18n.t("No pages found")}</TableCell>
          </TableRow>
        )}
      </TableBody>
    </Table>
  )
);

export default PageListComponent;
