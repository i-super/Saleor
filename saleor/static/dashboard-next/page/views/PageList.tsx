import Add from "material-ui-icons/Add";
import FilterListIcon from "material-ui-icons/FilterList";
import Card from "material-ui/Card";
import Drawer from "material-ui/Drawer";
import Grid from "material-ui/Grid";
import Hidden from "material-ui/Hidden";
import IconButton from "material-ui/IconButton";
import { stringify as stringifyQs } from "qs";
import * as React from "react";
import { Link } from "react-router-dom";

import ErrorMessageCard from "../../components/ErrorMessageCard";
import Navigator from "../../components/Navigator";
import PageHeader from "../../components/PageHeader";
import i18n from "../../i18n";
import PageFilters from "../components/PageFilters";
import PageListComponent from "../components/PageList";
import { pageEditUrl, pageStorefrontUrl } from "../index";
import { pageListQuery, TypedPageListQuery } from "../queries";

interface PageListProps {
  filters: any;
}

interface PageListState {
  isFilterMenuOpened: boolean;
}

export class PageList extends React.Component<PageListProps, PageListState> {
  state = { isFilterMenuOpened: false };
  handleFilterMenuOpen = () => {
    this.setState(prevState => ({
      isFilterMenuOpened: !prevState.isFilterMenuOpened
    }));
  };
  render() {
    const { filters } = this.props;
    return (
      <TypedPageListQuery
        query={pageListQuery}
        variables={{ first: 4 }}
        fetchPolicy="network-only"
      >
        {({ data, loading, error, fetchMore }) => {
          if (error) {
            return (
              <ErrorMessageCard
                message={i18n.t("Something went terribly wrong.")}
              />
            );
          }

          const loadNextPage = () => {
            if (loading) {
              return;
            }
            return fetchMore({
              updateQuery: (previousResult, { fetchMoreResult }) => {
                return {
                  ...fetchMoreResult,
                  pages: {
                    ...fetchMoreResult.pages,
                    pageInfo: {
                      ...fetchMoreResult.pages.pageInfo,
                      hasPreviousPage: true
                    }
                  }
                };
              },
              variables: {
                after: data.pages.pageInfo.endCursor,
                first: 3
              }
            });
          };
          const loadPreviousPage = () => {
            if (loading) {
              return;
            }
            return fetchMore({
              updateQuery: (previousResult, { fetchMoreResult, variables }) => {
                return {
                  ...fetchMoreResult,
                  pages: {
                    ...fetchMoreResult.pages,
                    pageInfo: {
                      ...fetchMoreResult.pages.pageInfo,
                      hasNextPage: true
                    }
                  }
                };
              },
              variables: {
                before: data.pages.pageInfo.startCursor,
                first: undefined,
                last: 12
              }
            });
          };

          return (
            <Navigator>
              {navigate => {
                const applyFilters = data => {
                  navigate(
                    `?${stringifyQs({ ...filters, ...data.formData })}`,
                    true
                  );
                };
                const clearFilters = () => navigate("?");
                return (
                  <Grid container spacing={16}>
                    <Grid item xs={12}>
                      <Grid container spacing={16}>
                        <Grid item xs={12} md={9}>
                          <Card>
                            <PageHeader
                              title={i18n.t("Pages", {
                                context: "title"
                              })}
                            >
                              <IconButton
                                disabled={loading}
                                onClick={() => navigate("/pages/add/")}
                              >
                                <Add />
                              </IconButton>
                              <Hidden mdUp>
                                <IconButton
                                  disabled={loading}
                                  onClick={this.handleFilterMenuOpen}
                                >
                                  <FilterListIcon />
                                </IconButton>
                              </Hidden>
                            </PageHeader>
                            <PageListComponent
                              pageInfo={loading ? null : data.pages.pageInfo}
                              pages={loading ? null : data.pages.edges}
                              handlePreviousPage={loadPreviousPage}
                              handleNextPage={loadNextPage}
                              editPageUrl={pageEditUrl}
                              showPageUrl={pageStorefrontUrl}
                              loading={loading}
                            />
                          </Card>
                        </Grid>
                        <Grid item xs={12} md={3}>
                          <Hidden smDown>
                            <PageFilters
                              handleSubmit={applyFilters}
                              handleClear={clearFilters}
                              formState={filters}
                            />
                          </Hidden>
                          <Drawer
                            open={this.state.isFilterMenuOpened}
                            onClose={this.handleFilterMenuOpen}
                            anchor="bottom"
                          >
                            <PageFilters
                              handleSubmit={applyFilters}
                              handleClear={clearFilters}
                              formState={filters}
                            />
                          </Drawer>
                        </Grid>
                      </Grid>
                    </Grid>
                  </Grid>
                );
              }}
            </Navigator>
          );
        }}
      </TypedPageListQuery>
    );
  }
}

export default PageList;
