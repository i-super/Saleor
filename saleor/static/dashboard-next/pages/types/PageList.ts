/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: PageList
// ====================================================

export interface PageList_pages_edges_node {
  __typename: "Page";
  /**
   * The ID of the object.
   */
  id: string;
  title: string;
  slug: string;
  isPublished: boolean;
}

export interface PageList_pages_edges {
  __typename: "PageCountableEdge";
  /**
   * The item at the end of the edge
   */
  node: PageList_pages_edges_node;
}

export interface PageList_pages_pageInfo {
  __typename: "PageInfo";
  /**
   * When paginating backwards, are there more items?
   */
  hasPreviousPage: boolean;
  /**
   * When paginating forwards, are there more items?
   */
  hasNextPage: boolean;
  /**
   * When paginating backwards, the cursor to continue.
   */
  startCursor: string | null;
  /**
   * When paginating forwards, the cursor to continue.
   */
  endCursor: string | null;
}

export interface PageList_pages {
  __typename: "PageCountableConnection";
  edges: PageList_pages_edges[];
  /**
   * Pagination data for this connection.
   */
  pageInfo: PageList_pages_pageInfo;
}

export interface PageList {
  /**
   * List of the shop's pages.
   */
  pages: PageList_pages | null;
}

export interface PageListVariables {
  first?: number | null;
  after?: string | null;
  last?: number | null;
  before?: string | null;
}
