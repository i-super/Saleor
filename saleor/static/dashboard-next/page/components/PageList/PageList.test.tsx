import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import PageList from "./";

const pageList = [
  {
    cursor: "12345678",
    node: {
      id: "abcd",
      isVisible: true,
      slug: "lorem-ipsum-dolor",
      title: "Lorem ipsum"
    }
  },
  {
    cursor: "12345678",
    node: {
      id: "abcde",
      isVisible: true,
      slug: "lorem-ipsum-consectetur",
      title: "Pellentesque habitant morbi tristique senectus"
    }
  },
  {
    cursor: "12345678",
    node: {
      id: "abcdef",
      isVisible: true,
      slug: "lorem-ipsum-amet",
      title: "Pellentesque metus turpis"
    }
  },
  {
    cursor: "12345678",
    node: {
      id: "abcdefg",
      isVisible: true,
      slug: "lorem-ipsum-sit",
      title: "Nulla blandit ut lectus at placerat"
    }
  }
];

describe("<PageList />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <MemoryRouter>
        <PageList
          handlePreviousPage={jest.fn()}
          handleNextPage={jest.fn()}
          onEditClick={jest.fn()}
          onShowPageClick={jest.fn()}
          pageInfo={{
            endCursor: "",
            hasNextPage: false,
            hasPreviousPage: false,
            startCursor: ""
          }}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when data is fully loaded", () => {
    const component = renderer.create(
      <MemoryRouter>
        <PageList
          handlePreviousPage={jest.fn()}
          handleNextPage={jest.fn()}
          pages={pageList}
          onEditClick={jest.fn()}
          onShowPageClick={jest.fn()}
          pageInfo={{
            endCursor: "",
            hasNextPage: false,
            hasPreviousPage: false,
            startCursor: ""
          }}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when category list is empty", () => {
    const component = renderer.create(
      <MemoryRouter>
        <PageList
          handlePreviousPage={jest.fn()}
          handleNextPage={jest.fn()}
          pages={[]}
          onEditClick={jest.fn()}
          onShowPageClick={jest.fn()}
          pageInfo={{
            endCursor: "",
            hasNextPage: false,
            hasPreviousPage: false,
            startCursor: ""
          }}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
