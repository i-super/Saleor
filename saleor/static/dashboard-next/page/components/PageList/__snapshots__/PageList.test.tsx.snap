// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageList /> renders when category list is empty 1`] = `
<table
  className="MuiTable-root-3"
>
  <thead
    className="MuiTableHead-root-4"
  >
    <tr
      className="MuiTableRow-root-5 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Url
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Visibility
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11 Component-textRight-2"
        scope="col"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-5 MuiTableRow-footer-7"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-footer-13 MuiTablePagination-root-21"
        colSpan={4}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-27 MuiToolbar-gutters-28 MuiTablePagination-toolbar-26"
        >
          <div
            className="MuiTablePagination-spacer-25"
          />
          <div
            className="MuiTablePaginationActions-root-29"
          >
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-45"
  >
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        colSpan={3}
        scope={undefined}
      >
        No pages found
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<PageList /> renders when data is fully loaded 1`] = `
<table
  className="MuiTable-root-3"
>
  <thead
    className="MuiTableHead-root-4"
  >
    <tr
      className="MuiTableRow-root-5 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Url
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Visibility
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11 Component-textRight-2"
        scope="col"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-5 MuiTableRow-footer-7"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-footer-13 MuiTablePagination-root-21"
        colSpan={4}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-27 MuiToolbar-gutters-28 MuiTablePagination-toolbar-26"
        >
          <div
            className="MuiTablePagination-spacer-25"
          />
          <div
            className="MuiTablePaginationActions-root-29"
          >
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-45"
  >
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-link-1"
        onClick={undefined}
        scope={undefined}
      >
        Lorem ipsum
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        /lorem-ipsum-dolor
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        Published
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-textRight-2"
        scope={undefined}
      >
        <button
          className="MuiButtonBase-root-36 MuiIconButton-root-30"
          disabled={false}
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-47"
          />
        </button>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-link-1"
        onClick={undefined}
        scope={undefined}
      >
        Pellentesque habitant morbi tristique senectus
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        /lorem-ipsum-consectetur
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        Published
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-textRight-2"
        scope={undefined}
      >
        <button
          className="MuiButtonBase-root-36 MuiIconButton-root-30"
          disabled={false}
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-47"
          />
        </button>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-link-1"
        onClick={undefined}
        scope={undefined}
      >
        Pellentesque metus turpis
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        /lorem-ipsum-amet
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        Published
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-textRight-2"
        scope={undefined}
      >
        <button
          className="MuiButtonBase-root-36 MuiIconButton-root-30"
          disabled={false}
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-47"
          />
        </button>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-link-1"
        onClick={undefined}
        scope={undefined}
      >
        Nulla blandit ut lectus at placerat
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        /lorem-ipsum-sit
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        Published
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-textRight-2"
        scope={undefined}
      >
        <button
          className="MuiButtonBase-root-36 MuiIconButton-root-30"
          disabled={false}
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-47"
          />
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<PageList /> renders while data is loading 1`] = `
<table
  className="MuiTable-root-3"
>
  <thead
    className="MuiTableHead-root-4"
  >
    <tr
      className="MuiTableRow-root-5 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Url
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11"
        scope="col"
      >
        Visibility
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-head-11 Component-textRight-2"
        scope="col"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-5 MuiTableRow-footer-7"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-footer-13 MuiTablePagination-root-21"
        colSpan={4}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-27 MuiToolbar-gutters-28 MuiTablePagination-toolbar-26"
        >
          <div
            className="MuiTablePagination-spacer-25"
          />
          <div
            className="MuiTablePaginationActions-root-29"
          >
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-35"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-39"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-45"
  >
    <tr
      className="MuiTableRow-root-5"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        <span
          className="Component-skeleton-46"
          style={undefined}
        >
          ‌
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        <span
          className="Component-skeleton-46"
          style={undefined}
        >
          ‌
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12"
        scope={undefined}
      >
        <span
          className="Component-skeleton-46"
          style={undefined}
        >
          ‌
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-10 MuiTableCell-body-12 Component-textRight-2"
        scope={undefined}
      >
        <button
          className="MuiButtonBase-root-36 MuiButtonBase-disabled-37 MuiIconButton-root-30 MuiIconButton-disabled-34"
          disabled={true}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
